source/_ext/gitstamp.py

```diff
@@ -0,0 +1,139 @@
+# Permission is hereby granted, free of charge, to any person obtaining a copy
+# of this software and associated documentation files (the "Software"), to deal
+# in the Software without restriction, including without limitation the rights
+# to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+# copies of the Software, and to permit persons to whom the Software is
+# furnished to do so, subject to the following conditions:
+#
+# The above copyright notice and this permission notice shall be included in
+# all copies or substantial portions of the Software.
+
+import os
+import datetime
+from sphinx import errors
+from github import Github
+
+# Gets the datestamp of the latest commit on the given file
+# Converts the datestamp into something more readable
+# Skips files whose datestamp we can't parse.
+# Expected git datestamp format: 2017-06-07 11:57:38 +1000
+# Output to June 7, 2017
+
+class GHCachedUser:
+    """
+    A user cached from GitHub
+    """
+    def __init__(self, name):
+        self.name = name
+        self.login = None
+        self.avatar_url = None
+
+def page_context_handler(app, pagename, templatename, context, doctree):
+    import git
```

Comment:
why importing here instead of at top of file?

Reply:
I *think* it's because it has been tried before (in `what_build_am_i`) and thrown the right exception if failed. At this point this import is guaranteed to succeed then.
If it was replaced by a single import at the top of the file we probably wouldn't have the chance to throw the exception and crash everything instead of just aborting this extension.
That's my guess though.